CMSgov/design-system
# [WNMGDS-2746][WNMGDS-2776] Update token build scripts to use JSON source

## Summary

Along with the Figma migration, we're implementing a new token storage method based on JSON ([see this pull request for background info](https://github.com/CMSgov/design-system/pull/2946)). Our previous scripts for exporting our token data into CSS, SCSS, CSV, and JSON are therefore outdated and need to be rewritten to take the new JSON format as input.

- Removed support for exporting tokens to CSV
    - This was helpful when we didn't have any other tools for viewing token data in table format, but now we have Figma's variable editor tables, and they are actually really handy. We can always re-implement this in the future if we need it.
- Removed support for exporting tokens to JSON
    - Our tokens are now natively in JSON, and we were previously exporting to this format to make it easier to import them into Sketch. We don't need to support that old Sketch code anymore.
- Rewrote the exporters for CSS and SCSS to use the JSON tokens as input
- Updated Figma-syncing code to support peculiarities with translating between CSS-based _dimensional_ units and Figma's generic _number_ type
- Updated the package README and included an explanation of our token structure from a high-level standpoint

This is a (pull) request to merge into a feature branch, as this feature is not completely ready to merge into the main branch.

### CSS changes to address:

Running `yarn test:unit -u packages/design-system-tokens` will show some differences between the new generated CSS-variable files and the ones generated by the previous tokens system. The following are outstanding changes to our CSS variables that either need to be accepted with VRT changes or need to be fixed in some way:

- [ ] Addition (or failure of removal) of `--autocomplete*` variables
- [ ] Change to CMS.gov badge radii
- [ ] Missing `--button__font-weight` in Figma
- [ ] Figure out how to use `--dropdown-option-group__padding` and `--dropdown__background-color` in the CSS
- [ ] Address all font family and font weight variable changes
- [ ] Use new `--global__color*` variables in the reset file
- [ ] Find out why `--label__*` variables disappeared
- [ ] Stop using `--shadow-base` in CSS
- [ ] Investigate missing `--usa-banner` variables

## How to test

1. `yarn build`
2. `yarn test:unit`
3. `yarn storybook`
4. Note that because of the list of CSS changes that still need to be addressed, running the VRTs is probably going to produce a large number of errors
5. Note also that the doc site does not build because it still references old token modules
6. Try adding new variables or deleting them in Figma and seeing if they show up after syncing
   - Note that this branch still references a testing Figma file rather than the true production file. I'll need to change this in the `.env.example` file in a future PR
